lib/pub_sub/testing/subscribe_to.rb

```diff
@@ -0,0 +1,36 @@
+RSpec::Matchers.define :subscribe_to do |event_name|
+  match do |domain|
+    handler_class = build_handler_class(event_name, domain)
+    event_class = build_event_class(event_name)
+    subscription_type = async? ? :async : :sync
+
+    expect(
+      PubSub::EventHandlerBuilder.new(handler_class, subscription_type)
+    ).to have_subscribed_to_events(event_class).in(event_store)
+  end
+
+  chain :asynchronously do
+    @asynchronously = true
+  end
+
+  private
+
+  def build_handler_class(event_name, domain)
+    handler_name = event_name.to_s.sub('__', '/').camelize
+    handler_name += 'Handler'
+    handler_name.remove!('::')
+    handler_name.prepend("#{domain.name}::")
+    handler_name.constantize
```

Comment:
Just like when building an event class, some simplifications can be applied:
```suggestion
    handler_name = event_name.to_s.sub('__', '/').camelize.remove('::')
    "#{domain.name}::#{handler_name}Handler".constantize
```

Reply:
The code you suggest is not an equivalent. But I did some changes based on your proposal :)